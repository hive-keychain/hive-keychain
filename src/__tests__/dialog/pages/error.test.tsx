import { render, waitFor } from '@testing-library/react';
import React from 'react';
import DialogError from 'src/dialog/pages/error';
import pagesMocks from 'src/__tests__/dialog/pages/mocks/pages-mocks';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import { clickAwait } from 'src/__tests__/utils-for-testing/setups/events';
import { PropsRequestDialogError } from 'src/__tests__/utils-for-testing/types/props-types';
describe('error tests:\n', () => {
  const { methods, spies } = pagesMocks;
  const props: PropsRequestDialogError = {
    data: { msg: { display_msg: 'Error on this Dialog!!!' } },
  };
  methods.config();
  it('Must show DialogError', async () => {
    const { asFragment } = render(<DialogError {...props} />);
    await waitFor(() => {});
    expect(asFragment()).toMatchSnapshot('DialogError');
  });
  it('Must call window.close when clicking on close', async () => {
    render(<DialogError {...props} />);
    await clickAwait([alButton.close]);
    expect(spies.closeWindow).toBeCalledTimes(1);
  });
});
