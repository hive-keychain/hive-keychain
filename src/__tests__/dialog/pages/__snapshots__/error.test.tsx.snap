// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error tests:
 Must show DialogError: DialogError 1`] = `
<DocumentFragment>
  <h2
    aria-label="dialog-header-error"
    class="dialog-header"
  >
    Error
  </h2>
  <p
    style="word-break: break-word;"
  >
    Error on this Dialog!!!
  </p>
  <button
    aria-label="button-close"
    class="submit-button default "
  >
    <div
      class="button-label"
    >
      Ok 
    </div>
  </button>
</DocumentFragment>
`;
